src/basic/strv.c

```diff
@@ -271,9 +267,57 @@ int strv_extend_strv(char ***a, char * const *b, bool filter_duplicates) {
         return -ENOMEM;
 }
 
+int strv_extend_strv_consume(char ***a, char **b, bool filter_duplicates) {
+        _cleanup_strv_free_ char **b_consume = b;
+        size_t p, q, i;
+
+        assert(a);
+
+        q = strv_length(b);
+        if (q == 0)
+                return 0;
+
+        p = strv_length(*a);
+        if (p >= SIZE_MAX - q)
+                return -ENOMEM;
+
+        char **t = reallocarray(*a, GREEDY_ALLOC_ROUND_UP(p + q + 1), sizeof(char *));
+        if (!t)
+                return -ENOMEM;
+        *a = t;
+
+        if (!filter_duplicates) {
+                assert(!size_multiply_overflow(sizeof(char *), q));
+                *mempcpy_typesafe(t + p, b, q * sizeof(char *)) = NULL;
```

Comment:
Hmm, please move the multiplier to `mempcpy_typesafe()`. Most (maybe all) of our typesafe macros internally multiply the size of type and take number of elements rather than the memory size.